### DISCUSSION TITLE
Display error on "Unknown Command" instead of showing usage.

### DISCUSSION THREAD

**Discussion by surgiie** [bashly-framework/bashly#243] [Ideas]
I typically see command line utilities throw a "unknown command" error and exit when a user tries to call a command that doesnt exist. Currently bashly will just show the usage command instead. Is there any plans to allow an error to be thrown instead or is there a way to accomplish this with Bashly's current features? 

So for example `my-cli unknown-cmd` would display an error: `Unknown command: "unknown-cmd"` and then exit.

**Comment by DannyBen** [comment]
Sure, good idea.

Do we need this to be configurable? Or just change the behavior for everybody?

**Reply by surgiie** [reply]
@DannyBen I dont have any preference. I think its fine for it to be the default.

**Comment by DannyBen** [comment]
I think I can implement it. I am also leaning towards default with no preferences settings.
Not sure why I did it differently.

Barring unforeseen complications, you can expect it in a few days.

**Comment by DannyBen** [comment]
This is tracked in https://github.com/DannyBen/bashly/issues/244
Should we add "Did you mean" suggestion on invalid commands?

**Reply by surgiie** [reply]
Im a fan of the "Did you mean" suggestion. If not too much trouble to add, I think its a worthwhile feature to implement.

**Reply by DannyBen** [reply]
The code was modified to show error instead of usage. It is not yet released, but if you need this quickly, you can use the version in master.

I am still toying with the *did you mean* functionality.

**Reply by surgiie** [reply]
Not a problem. I’m not in any hurry for this and will wait for the next release. Thank you for your work!

**Reply by adrian-gierakowski** [reply]
Just my 2 cents: it’s quite common amount cli tools to print usage after an error message so maybe bashly should do both? Or at least something like:

```
Error: Unknown command: "unknown-cmd"

Did you mean: “some-command”?

Run: “my-cli help” for list of all available commands.
```

**Reply by DannyBen** [reply]
Thanks, I have considered it, but for now I am leaning towards no. Rationale:

1. Why do we show usage on this error, and not other errors.
2. It should be obvious to everybody who has used a command line for more than one day that `--help` is showing help, and running the command without arguments shows usage.
3. If someone really wants this, [they can easily override one or all strings](https://bashly.dannyb.co/advanced/strings/).

**Comment by DannyBen** [comment]
Version 0.8.3 is released with this update. Thanks for the suggestion.
"Did you mean" did not make it in, but I am still considering it (I already have a working implementation, just not sure it is worth it).

**Reply by surgiie** [reply]
Thank you for your work.
